components/webhook-mgt/org.wso2.carbon.identity.webhook.metadata/src/main/java/org/wso2/carbon/identity/webhook/metadata/internal/dao/impl/FileBasedWebhookMetadataDAOImpl.java

```diff
@@ -142,13 +143,16 @@ private void loadEventProfiles() throws WebhookMetadataException {
     }
 
     @Override
-    public List<String> getSupportedEventProfiles() throws WebhookMetadataException {
+    public List<EventProfileMetadata> getSupportedEventProfiles() throws WebhookMetadataException {
```

Comment:
Rather than adding a new model, at the service layer return the event profiles loaded (these are pre loaded data as well). In the API return what need to be exposed only

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6798/commits/f3a36127565a733b73f51f7c9b4193df2120ca2e